src/libtiled/tilelayer.cpp

```diff
@@ -280,30 +285,29 @@ void TileLayer::flip(FlipDirection direction)
                 int _x = it.key().x() * CHUNK_SIZE + x;
                 int _y = it.key().y() * CHUNK_SIZE + y;
 
-                if (!contains(_x, _y))
-                    continue;
-
                 const Cell &source = it.value().cellAt(x, y);
                 Cell dest(source);
 
+                if (dest.isEmpty())
```

Comment:
You can do this check a little earlier based on `source`, but in fact I wonder why you have two variables at all. I'd just do:
```c++
Cell cell = it.value().cellAt(x, y);
if (cell.isEmpty())
    continue;
```